common/persistence/nosql/nosql_history_store_test.go

```diff
@@ -0,0 +1,172 @@
+		IsNewBranch: false,
+		Info:        "TestInfo",
+		BranchInfo: types.HistoryBranch{
+			TreeID:   "TestTreeID",
+			BranchID: "TestBranchID",
+			Ancestors: []*types.HistoryBranchRange{
+				{
+					BranchID:    "TestAncestorBranchID",
+					BeginNodeID: 0,
+					EndNodeID:   5,
+				},
+			},
+		},
+		NodeID: testNodeID,
+		Events: &persistence.DataBlob{
+			Encoding: common.EncodingTypeThriftRW,
+			Data:     []byte("TestEvents"),
+		},
+		TransactionID: testTransactionID,
+		ShardID:       testShardID,
+	}
+}
+
+func validHistoryNodeRow() *nosqlplugin.HistoryNodeRow {
+	expectedNodeRow := &nosqlplugin.HistoryNodeRow{
+		TreeID:       "TestTreeID",
+		BranchID:     "TestBranchID",
+		NodeID:       testNodeID,
+		TxnID:        common.Ptr[int64](123),
+		Data:         []byte("TestEvents"),
+		DataEncoding: string(common.EncodingTypeThriftRW),
+		ShardID:      testShardID,
+	}
+	return expectedNodeRow
+}
+
+func registerCassandraMock(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	mockDB := nosqlplugin.NewMockDB(ctrl)
+
+	mockPlugin := nosqlplugin.NewMockPlugin(ctrl)
+	mockPlugin.EXPECT().CreateDB(gomock.Any(), gomock.Any(), gomock.Any()).Return(mockDB, nil).AnyTimes()
+	RegisterPlugin("cassandra", mockPlugin)
+}
+
+func TestNewNoSQLHistoryStore(t *testing.T) {
+	registerCassandraMock(t)
+	cfg := getValidShardedNoSQLConfig()
+
+	_, err := newNoSQLHistoryStore(cfg, log.NewNoop(), nil)
+	assert.NoError(t, err)
+}
+
+func setUpMocks(t *testing.T) (*nosqlHistoryStore, *nosqlplugin.MockDB) {
+	ctrl := gomock.NewController(t)
+	dbMock := nosqlplugin.NewMockDB(ctrl)
+
+	nosqlSt := nosqlStore{
+		logger: log.NewNoop(),
+		db:     dbMock,
+	}
+
+	shardedNosqlStoreMock := NewMockshardedNosqlStore(ctrl)
+	shardedNosqlStoreMock.EXPECT().GetStoreShardByHistoryShard(testShardID).Return(&nosqlSt, nil).AnyTimes()
+
+	store := &nosqlHistoryStore{
+		shardedNosqlStore: shardedNosqlStoreMock,
+	}
+
+	return store, dbMock
+}
+
+func TestAppendHistoryNodes_ErrorIfAppendAbove(t *testing.T) {
+	store, _ := setUpMocks(t)
+
+	request := validInternalAppendHistoryNodesRequest()
+
+	// If the nodeID to append is smaller than the last ancestor's end node ID, return an error
+	request.NodeID = 3
+	ans := request.BranchInfo.Ancestors
+	ans[len(ans)-1].EndNodeID = 5
+
+	err := store.AppendHistoryNodes(ctx.Background(), request)
+
+	var invalidErr *persistence.InvalidPersistenceRequestError
+	assert.ErrorAs(t, err, &invalidErr)
+	assert.ErrorContains(t, err, "cannot append to ancestors' nodes")
+}
+
+func TestAppendHistoryNodes_NotNewBranch(t *testing.T) {
+	store, dbMock := setUpMocks(t)
+
+	// Expect to insert the node into the history tree and node, as this is not a new branch, expect treeRow to be nil
+	dbMock.EXPECT().InsertIntoHistoryTreeAndNode(gomock.Any(), nil, validHistoryNodeRow()).Return(nil).Times(1)
+
+	request := validInternalAppendHistoryNodesRequest()
+	err := store.AppendHistoryNodes(ctx.Background(), request)
+
+	assert.NoError(t, err)
+}
```

Comment:
Is there something else we could check in `store`? I wonder if we can check the amount of nodes is now == 1

Reply:
No, the main check is that we send the correct thing to the database. The historyStore struct does not store anything itself.